ISISNeutronMuon/MDANSE
# Store charge information in trajectory

**Description of work**
In some cases, the information about the partial charges is already contained in the MD engine files. This can be stored in the trajectory file and used for calculations of charge-related properties.

PR https://github.com/ISISNeutronMuon/MDANSE/pull/518 introduces a Trajectory Editor. That will have to be adjusted to handle charges once that PR is merged.

No attempt has been made to handle charges in CenterOfMassesTrajectory, since that one will require more fundamental changes (see https://github.com/ISISNeutronMuon/MDANSE/issues/545 ).

**Fixes**
1. Several (but not all) converters attempt to find charge information in the input files.
2. The charge values are stored in the MDT trajectory,
3. If the charge values change between frames, the information about charges is stored for each frame,\
4. Infrared analysis reads the charges from the trajectory file, **unless** the user specified charges using charge configurator.

**To test**
All tests must pass.
Please try converting a LAMMPS trajectory or DL_POLY trajectory which contains charges. Infrared analysis should produce a non-zero signal even when charges are not defined. Setting the charges using the charge configurator should change the results.
Analysis types which output a trajectory as a result should not discard the charge information.
Behaviour for trajectories without charge information should be the same as before.
